src/statics.rs

```diff
@@ -80,29 +87,98 @@ pub(crate) static EVENT_METADATA: once_cell::sync::Lazy<
 
 impl core::cmp::PartialEq for crate::_details::ParsedEventMetadata {
     fn eq(&self, other: &Self) -> bool {
-        self.identity_hash == other.identity_hash
+        match self.identity_hash == other.identity_hash {
+            true => self.meta.identity == other.meta.identity,
+            false => false
+        }
     }
 }
 
 impl core::cmp::Eq for crate::_details::ParsedEventMetadata {}
 
 impl core::cmp::PartialOrd for crate::_details::ParsedEventMetadata {
     fn partial_cmp(&self, other: &Self) -> Option<cmp::Ordering> {
-        self.identity_hash.partial_cmp(&other.identity_hash)
+        match self.identity_hash.partial_cmp(&other.identity_hash) {

```

Comment:
From the doc for PartialOrd:

> This trait should only contain the comparison logic for a type if one plans on only implementing PartialOrd but not [Ord](https://doc.rust-lang.org/std/cmp/trait.Ord.html). Otherwise the comparison logic should be in [Ord](https://doc.rust-lang.org/std/cmp/trait.Ord.html) and this trait implemented with Some(self.cmp(other)).

Reply:
Also:

> The methods of this trait must be consistent with each other and with those of [PartialEq](https://doc.rust-lang.org/std/cmp/trait.PartialEq.html).

Your implementation of Ord is not consistent with your implementation of PartialEq. Your PartialEq should return `Equal == self.cmp(other)`.